ssl/t1_lib.c

```diff
@@ -1318,7 +1318,7 @@ TICKET_RETURN tls_decrypt_ticket(SSL *s, const unsigned char *etick,
         /* Some additional consistency checks */
         if (p != sdec + slen || sess->session_id_length != 0) {
```

Comment:
I could set slen -= p-sdec immediately before the OPENSSL_free, and check that slen is 0 instead,
but to me it looks like overkill.
It is something fundamentally different to use a pointer to an automatic object that goes out of scope.
If the compiler optimizes the pointer variable away, then anything can happen, exactly as described
in 6.2.4.